socketio/src/client/client.rs

```diff
@@ -238,9 +242,22 @@ impl Client {
 
     fn callback<P: Into<Payload>>(&self, event: &Event, payload: P) -> Result<()> {
         let mut on = self.on.write()?;
+        let mut on_any = self.on_any.write()?;
         let lock = on.deref_mut();
+        let on_any_lock = on_any.deref_mut();
+
+        let payload = payload.into();
+
         if let Some(callback) = lock.get_mut(event) {
-            callback(payload.into(), self.clone());
+            callback(payload.clone(), self.clone());
+        }
+        match event.clone() {
+            Event::Message | Event::Custom(_) => {
+                if let Some(callback) = on_any_lock {
+                    callback(event.clone(), payload, self.clone())
+                }
+            }
+            _ => {}
         }
         drop(on);
```

Comment:
Either add a `drop(on_any)` here or remove this line, the lock is dropped on the next call anyways as it's residing on the stack.